core/src/main/java/tachyon/underfs/UnderFileSystem.java

```diff
@@ -134,6 +137,32 @@ protected UnderFileSystem(TachyonConf tachyonConf) {
     mTachyonConf = tachyonConf;
   }
 
+  /**
+   * Takes any necessary actions required to establish a connection to the under file system from
+   * the given master host e.g. logging in
+   * <p>
+   * Depending on the implementation this may be a no-op
+   * </p>
+   * 
+   * @param conf Tachyon configuration
+   * @param hostname The host that wants to connect to the under file system
+   * @throws IOException
+   */
+  public abstract void connectFromMaster(TachyonConf conf, String hostname) throws IOException;
```

Comment:
It's hard to tell the difference between connectFromMaster and connectFromWorker. Please create a JIRA to investigate a better way to do / name these two metnods.